crates/subspace-farmer/src/multi_farming.rs

```diff
@@ -179,6 +182,33 @@ impl MultiFarming {
             })
             .await?;
 
+            // Enable DSN subscription for segments archiving.
+            if enable_dsn_archiving {
```

Comment:
There are two things missing in this construction:
1) Disabling existing archiving mechanism (fine to not have this until actually implemented)
2) Archiving as implemented here with background task is not possible to interrupt, but we need that

Reply:
I would rather postpone both these tasks until the actual implementation. I can add some cancellation future in the background task but I'm not sure it will be the actual interface.